shapely/errors.py

```diff
@@ -43,8 +43,11 @@ class EmptyPartError(ShapelyError):
     """An error signifying an empty part was encountered when creating a multi-part."""
 
 
-class GeometryTypeError(ShapelyError):
+class GeometryTypeError(TypeError, ValueError):
     """
     An error raised when the type of the geometry in question is
-    unrecognized or inappropriate
+    unrecognized or inappropriate.
     """
+    def __init__(self, msg):
+        warnings.warn("GeometryTypeError will derive from ShapelyError and not TypeError or ValueError in Shapely 2.0.", ShapelyDeprecationWarning, stacklevel=2)
```

Comment:
@rgangopadhya looks like `import warnings` is needed at the top of errors.py.

Reply:
Sorry, yeah. Spent like 10 mins on this last week and then got distracted by other things -- this week